From fe1b6a02c8d2ff98c6d14919b239989cb2763d02 Mon Sep 17 00:00:00 2001
From: Pat Thoyts <patthoyts@users.sourceforge.net>
Date: Fri, 10 Sep 2010 10:40:03 +0100
Subject: [PATCH 3/3] curl: treat .netrc files as text when opening to handle
 line endings

The .netrc file may well use CRLF on Windows and should be opened using
text mode to avoid reading in the CR as part of the authentication data.

Signed-off-by: Pat Thoyts <patthoyts@users.sourceforge.net>

diff --git a/lib/netrc.c b/lib/netrc.c
index 7435d94..ea53e07 100644
--- a/lib/netrc.c
+++ b/lib/netrc.c
@@ -111,7 +111,7 @@ int Curl_parsenetrc(const char *host,
     netrc_alloc = TRUE;
   }
 
-  file = fopen(netrcfile, "r");
+  file = fopen(netrcfile, "rt");
   if(netrc_alloc)
     Curl_safefree(netrcfile);
   if(file) {
-- 
2.1.0

